### PR TITLE
utils_time: enable serial login

### PR DESCRIPTION
**Pull request by sitoliu**
Enable serial login in some cases and strip control code for
console session.

ID: 1649190

Signed-off-by: Sitong Liu <siliu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@sitoliu Please make sure each line not exceed 79 characters, thanks.

File: virttest/utils_time.py [hunk 1]
```diff
@@ -14,10 +14,11 @@ def get_host_timezone():
     """
     Get host's timezone
     """
-    timezone_cmd = 'timedatectl | grep "Time zone"'
+    timezone_cmd = 'timedatectl | /usr/bin/grep "Time zone"'
```
Comment (vivianQizhu) [inline virttest/utils_time.py]: This should be unnecessary.

File: virttest/utils_time.py [hunk 2]
```diff
@@ -161,10 +163,10 @@ def execute(cmd, timeout=360, session=None):
     :return: Command output string
     """
     if session:
-        ret = session.cmd_output(cmd, timeout=timeout)
+        ret = session.cmd_output_safe(cmd, timeout=timeout)
     else:
         ret = process.getoutput(cmd)
-    target = session and "guest" or "host"
+    target = 'guest' if session else 'host'
```
Comment (vivianQizhu) [inline virttest/utils_time.py]: Their logic are same. Unnecessary change.

File: virttest/utils_time.py (hunk 1 above)
Comment (sitoliu) [reply virttest/utils_time.py]: Thanks. Command `grep --color` may have alias name `grep` in some systems, so get explicit `grep` command with path.

Comment (sitoliu) [reply virttest/utils_time.py]: Thanks Qianqian, as we talked offline, this is mean to get explicit command 'grep' despite of the alias name.

File: virttest/utils_time.py (hunk 2 above)
Comment (sitoliu) [reply virttest/utils_time.py]: Thanks, yes, the logic is same, I update it as `pylint` suggest me to, so update it as well.

**Comment by sitoliu** [comment]
> @sitoliu This patch generally looks good to me, but would you detail each issue and how you fix them in the commit log separately, just as what you did in the bz(not everyone could see the content there). And I would also suggest you handle them in separate commit.

Thanks @vivianQizhu , updated.

File: virttest/utils_time.py [hunk 3]
```diff
@@ -105,20 +105,20 @@ def get_timezone_list():
 
     def get_timezone_code(timezone_name):
         for value in get_timezone_list():
-            if value[1] == timezone_name:
+            if len(value) == 2 and value[1] == timezone_name:
```
Comment (luckyh) [inline virttest/utils_time.py]: Well, the problem you met should be a bug of the serialport communication tool. It would be ok for me to make a workaround if we do not have enough time to fix it, just one question, does it work for network sessions?

Comment (luckyh) [inline virttest/utils_time.py]: And also, we'd better leave a comment to point out that it's just a workaround.

Comment (sitoliu) [reply virttest/utils_time.py]: Thanks. Updated, and patch works well for network session.

 (1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (804.80 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.timedrift.shared_ntp_date.with_reboot: PASS (138.13 s)

File: virttest/utils_time.py [hunk 4]
```diff
@@ -76,31 +76,24 @@ def verify_timezone_win(session):
...
+            if not code:
+                code = "+0000"
             else:
-                logging.warn("Can not get timezone name correctly.")
-            timezone_sets.append(timezone_set)
-            timezone_set = []
+                code = re.sub(r'(\d{2}):(\d{2})', r'\1\2', code)
+            codes.append(code)
+            names.append(name)
+        timezone_sets = list(zip(codes, names))
```
Comment (luckyh) [inline virttest/utils_time.py]: Suggest:
```python
match_pattern = "(?:\(UTC([+|-]\d{2}:\d{2})?)(?:.*\n)(\w+.*(?:\s\w+)*)"
timezone_sets = []
for para in re.split("(?:\r?\n){2,}", timezone_list.strip()):
    result = re.match(match_pattern, para, re.M).groups()
    if not result:
        continue
    code, name = result.groups()
    # When UTC standard time, add timezone code '+0000'
    if not code:
        code = "+0000"
    else:
        code = re.sub(r'(\d{2}):(\d{2})', r'\1\2', code)
    timezone_sets.append((code, name))
```

Comment (sitoliu) [reply virttest/utils_time.py]: Thanks, updated code, please help review it again, thanks.

Test pass.
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.timedrift.shared_ntp_date.with_reboot: PASS (349.02 s)

**Comment by luckyh** [comment]
Thanks all.
